File: grpc-example/grpc-example-java/hello-impl/src/main/java/com/example/hello/impl/HelloGrpcServiceImpl.java
package com.example.hello.impl;

import akka.actor.ActorSystem;
import akka.stream.Materializer;
import example.myapp.helloworld.grpc.AbstractGreeterServiceRouter;
import example.myapp.helloworld.grpc.HelloReply;
import example.myapp.helloworld.grpc.HelloRequest;

import javax.inject.Inject;
import javax.inject.Singleton;
import java.util.concurrent.CompletableFuture;
import java.util.concurrent.CompletionStage;

@Singleton
public class HelloGrpcServiceImpl extends AbstractGreeterServiceRouter {

    @Inject
    public HelloGrpcServiceImpl(ActorSystem sys, Materializer mat) {
        super(mat, sys);
    }

    @Override
    public CompletionStage<HelloReply> sayHello(HelloRequest in) {
        HelloReply reply = HelloReply
            .newBuilder()
            .setMessage("Hi " + in.getName() + " (gRPC)")
            .build();
        return CompletableFuture.completedFuture(reply);
    }
}

File: grpc-example/grpc-example-scala/hello-impl/src/main/scala/com/example/hello/impl/HelloGrpcServiceImpl.scala
package com.example.hello.impl

import akka.actor.ActorSystem
import example.myapp.helloworld.grpc.{ AbstractGreeterServiceRouter, HelloReply, HelloRequest }

import scala.concurrent.Future

class HelloGrpcServiceImpl(system: ActorSystem)
  extends AbstractGreeterServiceRouter(system) {
  override def sayHello(in: HelloRequest): Future[HelloReply] =
    Future.successful(HelloReply(s"Hi ${in.name}! (gRPC)"))
}
